skbio/tree/_tree.py

```diff
@@ -3650,168 +4149,540 @@ def update_result():
+
+        This method automatically determines whether to use the unrooted or rooted
+        RF distance based on to whether self is rooted (see :meth:`details <unroot>`).
+        This can be overridden by the ``rooted`` parameter, which is recommended for
+        explicity.
 
-        tree1_sets = tree1.subsets()
-        tree2_sets = tree2.subsets()
+        By specifying ``proportion=True``, a unit distance will be returned, ranging
+        from 0 (identical) to 1 (completely different).
 
-        not_in_both = tree1_sets.symmetric_difference(tree2_sets)
+        Only taxa shared between the two trees are considered. Taxa unique to either
+        tree are excluded from the calculation.
 
-        dist = float(len(not_in_both))
+        See Also
+        --------
+        compare_wrfd
+        compare_subsets
+        compare_biparts
+        compare_tip_distances
+
+        References
+        ----------
+        .. [1] Robinson, D. F., & Foulds, L. R. (1981). Comparison of phylogenetic
+           trees. Mathematical biosciences, 53(1-2), 131-147.
+
+        .. [2] Bogdanowicz, D., & Giaro, K. (2013). On a matching distance between
+           rooted phylogenetic trees. International Journal of Applied Mathematics
+           and Computer Science, 23(3), 669-684.
+
+        Examples
+        --------
+        Calculate the RF distance between two unrooted trees with the same taxa but
+        different topologies. Each tree has three non-trivial bipartitions, as defined
+        by individual internal branches, among which one pair (abc|def) is shared
+        whereas the other two of each tree are unique (ab|cdef, abcf|de, bc|adef,
+        abcd|ef). Therefore the RF distance is 2 + 2 = 4.
+
+        >>> from skbio import TreeNode
+        >>> tree1 = TreeNode.read(["((a,b),c,((d,e),f));"])
+        >>> print(tree1.ascii_art())
+                            /-a
+                  /--------|
+                 |          \-b
+                 |
+        ---------|--c
+                 |
+                 |                    /-d
+                 |          /--------|
+                  \--------|          \-e
+                           |
+                            \-f
+
+        >>> tree2 = TreeNode.read(["((a,(b,c)),d,(e,f));"])
+        >>> print(tree2.ascii_art())
+                            /-a
+                  /--------|
+                 |         |          /-b
+                 |          \--------|
+                 |                    \-c
+        ---------|
+                 |--d
+                 |
+                 |          /-e
+                  \--------|
+                            \-f
 
-        if proportion:
-            total_subsets = len(tree1_sets) + len(tree2_sets)
-            dist /= total_subsets
+        >>> tree1.compare_rfd(tree2)
+        4
 
-        return dist
+        """
+        if rooted is None:
+            rooted = len(self.children) == 2
```

Comment:
```suggestion
            rooted = len(self.root().children) == 2
```

Reply:
...or would it be accurate for a subtree to be considered rooted/unrooted?

Reply:
This is a good point. scikit-bio was vague about whether a method affects the subtree or the entire tree. In some of the method I revised, I added a sentence like "This method modifies the subtree under the current node." to the docstring. But not all of them has that.

I designed the mechanism of deeming a tree is rooted or not based on whether it has two children. But I indeed overlooked the point you raised, which is a good point. To fix this, we need another look into the rooted/unrooted mechanism, and revise the documentation.